docs/reference/aggregations.asciidoc

```diff
@@ -330,79 +330,65 @@ the aggregated field.
 [[use-scripts-in-an-agg]]
 === Use scripts in an aggregation
 
-Some aggregations support <<modules-scripting,scripts>>. You can
-use a `script` to extract or generate values for the aggregation:
+When a field doesn't exactly match the aggregation you need to do you
+should aggregate on a <<runtime,runtime field>>:
 
 [source,console]
 ----
-GET /my-index-000001/_search
+GET /my-index-000001/_search?size=0
 {
+  "runtime_mappings": {
+    "message.length": {
+      "type": "long",
+      "script": "emit(doc['message.keyword'].value.length())"
+    }
+  },
   "aggs": {
-    "my-agg-name": {
+    "message_length": {
       "histogram": {
-        "interval": 1000,
-        "script": {
-          "source": "doc['my-field'].value.length()"
-        }
+        "interval": 10,
+        "field": "message.length"
       }
     }
   }
 }
 ----
 // TEST[setup:my_index]
-// TEST[s/my-field/http.request.method/]
 
-If you also specify a `field`, the `script` modifies the field values used in
-the aggregation. The following aggregation uses a script to modify `my-field`
-values:
-
-[source,console]
+////
+[source,console-result]
 ----
-GET /my-index-000001/_search
 {
-  "aggs": {
-    "my-agg-name": {
-      "histogram": {
-        "field": "my-field",
-        "interval": 1000,
-        "script": "_value / 1000"
-      }
+  "timed_out": false,
+  "took": "$body.took",
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0,
+    "skipped": 0
+  },
+  "hits": "$body.hits",
+  "aggregations": {
+    "message_length": {
+      "buckets": [
+        {
+          "key": 30.0,
+          "doc_count": 5
+        }
+      ]
     }
   }
 }
 ----
-// TEST[setup:my_index]
-// TEST[s/my-field/http.response.bytes/]
-
-Some aggregations only work on specific data types. Use the `value_type`
-parameter to specify a data type for a script-generated value or an unmapped
-field. `value_type` accepts the following values:
+////
 
-* `boolean`
-* `date`
-* `double`, used for all floating-point numbers
-* `long`, used for all integers
-* `ip`
-* `string`
+Scripts will calculate the values on the fly, adding a little overhead
+to the execution of the aggretion. Some aggregations like
+<<search-aggregations-bucket-terms-aggregation,`terms`>> and
+<<search-aggregations-bucket-filters-aggregation,`filters`>> have to
+give up some optimizations.
```

Comment:
```suggestion
Scripts calculate field values dynamically, which adds overhead
to the aggregation. Certain aggregations like
<<search-aggregations-bucket-terms-aggregation,`terms`>> and
<<search-aggregations-bucket-filters-aggregation,`filters`>> are
less optimized when using scripts.
```

Reply:
@nik9000, let me know if I totally misinterpreted your meaning here.

Reply:
Yeah, this is right. I think I meant something a little more nuanced with "a little" than comes across on rereading. I'll merge you suggestion and then iterate on it locally for a bit.